File: bewerberpool/frmLogin.vb
Imports System.Security
Imports System.Security.Cryptography
Imports System.IO
Imports System.Text
Imports bewerberpool.benutzerverwaltungDataSet
Public Class frmLogin

    Private _frmMain As frmMain

    Sub New(frmMain As frmMain)
        ' TODO: Complete member initialization 
        _frmMain = frmMain
        InitializeComponent()
    End Sub

    Private Sub frmLogin_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.BenutzerTableAdapter.Fill(Me.BenutzerverwaltungDataSet.benutzer)
        Me.KeyTableAdapter.Fill(Me.BenutzerverwaltungDataSet.key)
        Dim encryptionrow = DirectCast(DirectCast(KeyBindingSource.Current, DataRowView).Row, keyRow)
        crypto.Encryptionkey = CStr(encryptionrow.keycol)
    End Sub

    Public Function GetBenutzernameExists(ByVal benutzername As String) As Boolean
        Dim test1 As String = CStr(crypto.Encrypt(txtUsername.Text))
        Return Me.BenutzerverwaltungDataSet.benutzer.Any(Function(rw) rw.benutzername = test1)
    End Function

    Public Function GetPasswordExists(ByVal benutzername As String) As Boolean
        Dim test2 As String = CStr(crypto.Encrypt(txtPassword.Text))
        Return Me.BenutzerverwaltungDataSet.benutzer.Any(Function(rw) rw.kennwort = test2)
    End Function

    Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click

        usernameklar = CStr(txtUsername.Text)
        Dim username As String = CStr(crypto.Encrypt(txtUsername.Text))
        Dim password As String = CStr(crypto.Encrypt(txtPassword.Text))
        Dim vorhanden1 As Boolean = GetBenutzernameExists(username)
        Dim vorhanden2 As Boolean = GetPasswordExists(password)

        If String.IsNullOrEmpty(username) Then Exit Sub ' Ausführung anhalten, wenn nichts eingetragen wurde
        If String.IsNullOrEmpty(password) Then Exit Sub ' Ausführung anhalten, wenn nichts eingetragen wurde

        If vorhanden1 AndAlso vorhanden2 Then
            Me.BenutzerTableAdapter.ScalarQuery(Me.BenutzerverwaltungDataSet.benutzer, username, password)
            Dim loginflagausdataset = DirectCast(DirectCast(Me.BenutzerBindingSource.Current, DataRowView).Row, benutzerRow)
            loginflag = CInt(loginflagausdataset.zugang_bewpool)
            Me.Close()
        Else
            MsgBox("Username und/oder Passwort ist falsch.")
            loginflag = 0
            frmMain.Close()
            Me.Close()
        End If
    End Sub

    Private Sub btnAbbrechen_Click(sender As Object, e As EventArgs) Handles btnAbbrechen.Click
        frmMain.Close()
        Me.Close()
    End Sub

    Private Sub frmLogin_Shown(sender As Object, e As EventArgs) Handles Me.Shown
        AcceptButton = btnOK
    End Sub
End Class

File: bewerberpool/frmLogin.cs
using System.Threading.Tasks;
using System.Drawing;
using System.Data;
using System.Diagnostics;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using System.Linq;
using System.Collections.Generic;
using System.Collections;
using System;
using System.Security;
using System.Security.Cryptography;
using System.IO;
using System.Text;
using bewerberpool.benutzerverwaltungDataSet;
using Microsoft.VisualBasic.CompilerServices;

namespace bewerberpool
{
    public partial class frmLogin
    {
        private frmMain _frmMain;

        public frmLogin(frmMain frmMain)
        {
            // TODO: Complete member initialization 
            _frmMain = frmMain;
            InitializeComponent();
        }

        private void frmLogin_Load(object sender, EventArgs e)
        {
            this.BenutzerTableAdapter.Fill(this.BenutzerverwaltungDataSet.benutzer);
            this.KeyTableAdapter.Fill(this.BenutzerverwaltungDataSet.key);
            var encryptionrow = (benutzerverwaltungDataSet.keyRow)((DataRowView)KeyBindingSource.Current).Row;
            crypto.Encryptionkey = System.Convert.ToString(encryptionrow.keycol);
        }

        public bool GetBenutzernameExists(string benutzername)
        {
            string test1 = System.Convert.ToString(crypto.Encrypt(txtUsername.Text));
            return this.BenutzerverwaltungDataSet.benutzer.Any(rw => Operators.CompareString(rw.benutzername, test1, TextCompare: false) == 0);
        }

        public bool GetPasswordExists(string benutzername)
        {
            string test2 = System.Convert.ToString(crypto.Encrypt(txtPassword.Text));
            return this.BenutzerverwaltungDataSet.benutzer.Any(rw => Operators.CompareString(rw.kennwort, test2, TextCompare: false) == 0);
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            Module1.usernameklar = System.Convert.ToString(txtUsername.Text);
            string username = System.Convert.ToString(crypto.Encrypt(txtUsername.Text));
            string password = System.Convert.ToString(crypto.Encrypt(txtPassword.Text));
            bool vorhanden1 = GetBenutzernameExists(username);
            bool vorhanden2 = GetPasswordExists(password);

            if (string.IsNullOrEmpty(username))
                return; // Ausführung anhalten, wenn nichts eingetragen wurde
            if (string.IsNullOrEmpty(password))
                return; // Ausführung anhalten, wenn nichts eingetragen wurde

            if (vorhanden1 && vorhanden2)
            {
                this.BenutzerTableAdapter.ScalarQuery(this.BenutzerverwaltungDataSet.benutzer, username, password);
                var loginflagausdataset = (benutzerverwaltungDataSet.benutzerRow)((DataRowView)this.BenutzerBindingSource.Current).Row;
                Module1.loginflag = System.Convert.ToInt32(loginflagausdataset.zugang_bewpool);
                this.Close();
            }
            else
            {
                Interaction.MsgBox("Username und/oder Passwort ist falsch.");
                Module1.loginflag = 0;
                bewerberpool.My.MyProject.MyForms.frmMain.Close();
                this.Close();
            }
        }

        private void btnAbbrechen_Click(object sender, EventArgs e)
        {
            bewerberpool.My.MyProject.MyForms.frmMain.Close();
            this.Close();
        }

        private void frmLogin_Shown(object sender, EventArgs e)
        {
            AcceptButton = btnOK;
        }
    }
}
